jujugui/static/gui/src/app/components/account/credentials/credentials.js

```diff
@@ -0,0 +1,218 @@
+    },
+
+    /**
+      Handle deleting a credential.
+
+      @method _handleDeleteCredential
+      @param credential {String} A credential id.
+    */
+    _handleDeleteCredential: function(credential) {
+      const xhr = this.props.revokeCloudCredential(credential, (error) => {
+        if (error) {
+          const message = 'Unable to revoke the cloud credential';
+          this.props.addNotification({
+            title: message,
+            message: `${message}: ${error}`,
+            level: 'error'
+          });
+          console.error(message, error);
+          return;
+        }
+        // Remove the credential from the list.
+        const credentials = this.state.credentials.filter(cred => {
+          if (cred.name !== credential) {
+            return true;
+          }
+        });
+        this.setState({credentials: credentials});
+      });
+      this.xhrs.push(xhr);
+    },
+
+    /**
+      Generate a list of credentials.
+
+      @method _generateCredentials
+    */
+    _generateCredentials: function() {
+      const credentials = this.state.credentials;
+      if (this.state.loading) {
+        return (
+          <juju.components.Spinner />);
+      }
+      const credentialsList = credentials.map(credential => {
+        const cloud = this.props.getCloudProviderDetails(credential.cloud);
+        return (
+          <li className="user-profile__list-row twelve-col"
+            key={credential.name}>
+              <div className="six-col no-margin-bottom">
+                {credential.name}
+              </div>
+              <div className="four-col no-margin-bottom">
+                {cloud.title}
+              </div>
+              <div className="two-col last-col no-margin-bottom">
+                <juju.components.GenericButton
+                  action={
+                    this._handleDeleteCredential.bind(this, credential.name)}
+                  type="neutral"
+                  title="Remove" />
+              </div>
+          </li>);
+      });
+      if (credentialsList.length > 0) {
+        return (
+          <ul className="user-profile__list twelve-col">
+            <li className="user-profile__list-header twelve-col">
+              <div className="six-col no-margin-bottom">
+                Name
+              </div>
+              <div className="six-col last-col no-margin-bottom">
+                Provider
+              </div>
+            </li>
+            {credentialsList}
+          </ul>);
+      } else {
+        return (
+          <div>
+            No credentials available.
+          </div>);
+      }
+    },
+
+    render: function() {
+      return (
+        <div className="account__section">
+          <h2 className="account__title twelve-col">
+            Cloud credentials
+          </h2>
+          {this._generateCredentials()}
+        </div>
+      );
+    }
+
+  });
+
+}, '', {
+  requires: [
+    'account-payment-method-card',
+    'expanding-row',
```

Comment:
I don't think that the `account-payment-method-card` and `expanding-row` components are used here.